main/compiling/Circuit.py

```diff
@@ -427,19 +424,48 @@ def _to_stim(
         self.measurer.reset_compilation()
         return full_circuit
 
-    def instruction_to_stim(self, instruction: Instruction, circuit: stim.Circuit):
-        """Adds an individual Instruction to a circuit
+    def split_instructions_according_to_gate(self, qubit_instructions: Dict[Qubit, List[Instruction]]):
+        """Splits the instructions into gates and measurements
+
+        Qubit instructions is a dictionary whose keys are Qubits and values are lists of Instructions.
+        This function generates a dictionary whose keys are a instruction and whose values are lists of qubits.
+        This is done to reduce the amount of times we need to call Stim's append function.
+
+        It also generates a list of measurements.
 
         Args:
-            instruction: The instruction to be translated
-            circuit: A stim circuit to add the instruction to.
-        """
-        if instruction.targets is not None:
-            targets = instruction.targets
-        else:
-            targets = [self.qubit_index(qubit) for qubit in instruction.qubits]
-        circuit.append(instruction.name, targets, instruction.params)
+            qubit_instructions: A dictionary whose keys are Qubits and values are lists of Instructions.
 
+        Returns:
+            A tuple of two dictionaries. The first dictionary has keys of Instructions and values of lists of qubits.
+            The second dictionary has keys of Instructions which are measurements and values of lists of qubits.        
+        """
+        measurements = []
+        qubits_by_instruction = {}
+        compiled_instruction = defaultdict(bool)
+        instructions = sorted(qubit_instructions.values(),
+                              key=lambda value: (value[0].name, self.qubit_index(value[0].qubits[0])))
```

Comment:
Why value[0]? What about other value[j] for j > 0?

Reply:
updated the sorting function to loop through the values